src/components/main-grid/index.js

```diff
@@ -0,0 +1,35 @@
+// @vendors
+import { array, string } from "prop-types";
+
+// @components
+import GridITem from "./grid-item";
+
+// @styles
+import styles from "./styles.module.scss";
+
+const MainGrid = ({ data, title }) => {
+  return (
+    <div className={styles.mainContainer}>
+      <h3>{title}</h3>
+      <div className={styles.container}>
+        {data?.map((item) => (
+          <div key={item.id}>
+            <GridITem data={item.data} />
+          </div>
+        ))}
+      </div>
+    </div>
+  );
+};
+
+MainGrid.propTypes = {
+  data: array.isRequired,
+  title: string,
+};
+
+MainGrid.defaultProps = {
+  data: [],
+  title: "Our Products",
```

Comment:
nit: It feels like you would want to make the title explicit and not default to "Our Products", you could make title required